api/controllers/user.js

```diff
@@ -52,5 +52,15 @@ module.exports = {
             res.status(statusCodes.InternalServerError).json({ message: e.message });
             next();
         }
+    },
+    addToBasket: async (req, res, next) => {
```

Comment:
You don't need endpoint for basket on backend. You can store the products into cookies on the front-end.